vajram/vajram-codegen/src/main/java/com/flipkart/krystal/vajram/codegen/Utils.java

```diff
@@ -0,0 +1,449 @@
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.MirroredTypeException;
+import javax.lang.model.type.PrimitiveType;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.ElementFilter;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.SimpleTypeVisitor14;
+import javax.lang.model.util.Types;
+import javax.tools.Diagnostic.Kind;
+import javax.tools.JavaFileObject;
+
+public class Utils {
+
+  private static final boolean DEBUG = false;
+
+  public static final String DOT = ".";
+  public static final String COMMA = ",";
+  public static final String REQUEST_SUFFIX = "Request";
+  public static final String IMPL = "Impl";
+  public static final String INPUT_UTIL = "InputUtil";
+  public static final String CONVERTER = "CONVERTER";
+
+  private final ProcessingEnvironment processingEnv;
+  private final Types typeUtils;
+  private final Elements elementUtils;
+
+  public Utils(ProcessingEnvironment processingEnv) {
+    this.processingEnv = processingEnv;
+    this.typeUtils = processingEnv.getTypeUtils();
+    this.elementUtils = processingEnv.getElementUtils();
+  }
+
+  List<TypeElement> getVajramClasses(RoundEnvironment roundEnv) {
+    return roundEnv.getElementsAnnotatedWith(VajramDef.class).stream()
+        .filter(element -> element.getKind() == ElementKind.CLASS)
+        .map(executableElement -> (TypeElement) executableElement)
+        .toList();
+  }
+
+  VajramCodeGenerator createCodeGenerator(VajramInfo vajramInfo) {
+    Map<VajramID, VajramInfoLite> map = new HashMap<>();
+    for (DependencyModel depModel : vajramInfo.dependencies()) {
+      map.put(
+          depModel.depVajramId(),
+          new VajramInfoLite(depModel.depVajramId().vajramId(), depModel.responseType()));
+    }
+    return new VajramCodeGenerator(vajramInfo, map, processingEnv);
+  }
+
+  void generateSourceFile(String className, String code, TypeElement vajramDefinition) {
+    try {
+      JavaFileObject requestFile =
+          processingEnv.getFiler().createSourceFile(className, vajramDefinition);
+      note("Successfully Create source file %s".formatted(className));
+      try (PrintWriter out = new PrintWriter(requestFile.openWriter())) {
+        out.println(code);
+      }
+    } catch (FilerException e) {
+      // Since we do multiple passes (codeGenVajramModels, and compileJava) where this annotation
+      // processor is executed, we might end up creating the same file multiple times. This is not
+      // an error, hence we ignore this exceeption
+      //      note("Could not create source file for %s. Due to exception %s".formatted(className,
+      // e));
+    } catch (Exception e) {
+      error(
+          "Error creating java file for className: %s. Error: %s".formatted(className, e),
+          vajramDefinition);
+    }
+  }
+
+  public VajramInfo computeVajramInfo(TypeElement vajramClass) {
```

Comment:
We should add test cases for these utility methods.

Reply:
Since this is using compiler internals, it is hard to test these methods individually. 

1. The best way to test these is to test them end to end by writing vajrams which are processed by annotation processors which internally call these methods. (See: https://stackoverflow.com/a/18620422/2173948)

The `vajram-samples` module serves this purpose. All the vajrams in that module are essentially integration tests for the code generator.

In future, if we find some scenarios which cannot be tested by the above integ-testing method, we can invest in how to make compile time java lang models to test this method. This is my thought process currently. 
Does that sound ok?

Reply:
Got it.